src/clp_ffi_js/constants.hpp

```diff
@@ -17,6 +18,7 @@ enum class LogLevel : std::uint8_t {
     WARN,
     ERROR,
     FATAL,
+    LENGTH,  // This isn't a valid log level.
```

Comment:
I see what you're saying. On the other hand, having two invalid values is kind of messy already. For example, the code to check if a log level is within the valid enum values depends on `cValidLogLevelsBeginIdx` and `LogLevel::LENGTH`.

My perspective on the current log level parsing is that it's a heuristic until we migrate to IRv2 entirelyl; at that point, we shouldn't have to parse log levels any more. So I feel like simplifying the current code is a better approach until we switch to IRv2 or we have a user-need for more complicated levels.

That said, this isn't that big of a deal, so I guess if @junhaoliao doesn't agree, I'll leave it be.

Reply:
> the current log level parsing is that it's a heuristic until we migrate to IRv2 entirely

@kirkrodrigues Would CLP FFI also support parsing log level for IRv1 stream in the new `Deserializer` interface (the one that is now only for IRv2)? If not, we still need to maintain this enum for IRv1 till we completely drop support for IRv1, right?

Reply:
Yeah, we need to wait until we completely drop support for IRv1.

Reply:
I agree with @davemarco 's (1.) argument about the challenges of adding new log levels. However, I also recognize that the `NONE` value is used solely for log parsing (in read paths, not write paths). Since `NONE` is never serialized into non-code spaces and read back into the code, as long as our implementation consistently refers to it, there should be no risk of errors, which aligns with @kirkrodrigues's argument, assuming I understand it correctly.

With that said, I still prefer that we keep the `NONE` as the very first item in the enum.
The underlying type of the `LogLevel` is `uint8_t` which has value `0` when unutilized and placed in bss / initialized without explicit value (`={}`). Such value is usually used to hint the default; for example, in order to hint feature XXX is by default enabled, we can add Boolean flags like `is_xxx_disabled`. With `NONE` removed, when we declare a enum variable without proper initialization, it may contain a valid value, and most importantly the default value is no longer hinted to code readers. (Though most of the time we shouldn't really read before initialization lol

Reply:
In a way I feel the inline comments seem a bit bloated. Instead, as a new convention, can we add a prefix to hint the enum member is for describing properties of the enum? e.g., 
```
enum class FilesTableCol: std::uint8_t {
   NAME = 0,
   PATH,
   SIZE,
   FILES_TABLE_COL_LENGTH // then this should not be confused with `SIZE`
}
```

Reply:
* Sure, let's keep `NONE`.
* Sure, I guess we could define a new convention. Keep in mind the naming of enum values should be PascalCase.